tracer/src/Datadog.Trace/Debugger/LiveDebugger.cs

```diff
@@ -280,6 +284,89 @@ internal void UpdateAddedProbeInstrumentations(IReadOnlyList<ProbeDefinition> ad
             }
         }
 
+        private static string[] SafeParseSignature(string signature)
+        {
+            if (string.IsNullOrEmpty(signature))
+            {
+                return null;
+            }
+
+            try
+            {
+                return ParseSignature(signature);
+            }
+            catch (Exception e)
+            {
+                Log.Error(e, "Error parsing the signature {Signature}. Fall back to empty signature", signature);
+                return null;
+            }
+        }
+
+        private static string[] ParseSignature(string signature)
+        {
+            if (signature.StartsWith("(") && signature.EndsWith(")"))
+            {
+                signature = signature.Substring(1, signature.Length - 2);
+            }
+
+            var types = new List<string>();
+            var initialCapacity = 32;
+            var currentType = StringBuilderCache.Acquire(initialCapacity);
+            var stack = new Stack<char>();
+
+            try
+            {
+                foreach (char c in signature)
+                {
+                    if (c == ' ')
+                    {
+                        continue;
+                    }
+
+                    if (c == ',' && stack.Count == 0)
+                    {
+                        types.Add(StringBuilderCache.GetStringAndRelease(currentType));
+                        currentType = StringBuilderCache.Acquire(initialCapacity);
+                    }
+                    else
+                    {
+                        if (c == '<')
+                        {
+                            stack.Push(c);
+                            currentType.Append('['); // Replace < with [
+                        }
+                        else if (c == '>')
+                        {
+                            if (stack.Count > 0 && stack.Peek() == '<')
```

Comment:
AFAICT, you only _ever_ push `'<'` to the stack, so you can simplify all this to be an `int` that checks the "depth"?

Reply:
yep, refactoring this block of code significantly

Reply:
Done in a940eb08aea2701523ff68529684eff12fe361fd.